test/functorch/test_control_flow.py

```diff
@@ -2011,13 +2117,196 @@ def chain_fct_different_dim(inp):
             ),
             0,
+            reverse=reverse_second,
+        )
+        result = fct_cmp(inp)
+        self.assertEqual(result, expected_result)
+        
+    # TODO: provide an implementation for all compile modes and re-enable all test
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    # @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("combine_mode", ["pointwise"])
+    # @parametrize("compile_mode", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("compile_mode", ["none"])
+    @parametrize("reverse_first", [False, True])
+    @parametrize("same_direction", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping the combination of combine_mode=pointwise and device=cpu
+    # as the current implementation of pointwise does only support CUDA device
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
+            and (params["device"] == torch.device("cpu") or torch.version.hip)
+        ),
+    )
+    def test_associative_scan_nested(
+        self, combine_mode, compile_mode, reverse_first, same_direction, device
+    ):
+        reverse_second = reverse_first if same_direction else not reverse_first
+        scan_fct = compile_mode_helper(associative_scan, compile_mode)
+
+        def nested_associative_scan(x, y):
+            def inner_fn(xi, yi):
+                return xi * yi
+            
+            y_new = scan_fct(
+                inner_fn,
+                y,
+                0,
+                combine_mode=combine_mode,
+                reverse=reverse_second,
+            )
+            return x + y_new
+
+        inp = torch.randn(3, 10, 2, device=device)
+        
+        result = scan_fct(nested_associative_scan, inp, 0, reverse=reverse_first, combine_mode="generic")
+        # expected_result = _fake_associative_scan(
+        #     nested_associative_scan,
+        #     inp,
+        #     0,
+        #     reverse=reverse_first
+        # )
+        
+        # self.assertEqual(result, expected_result)
+        
+    # TODO: provide an implementation for all compile modes and re-enable all test
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("compile_mode", ["none", "compile", "compile_dynamic_shape"])
+    # @parametrize("loop_type", ["for", "while"])
+    @parametrize("loop_type", ["for"])
+    # @parametrize("loop_type", ["while"])
+    @parametrize("reverse", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping the combination of combine_mode=pointwise and device=cpu
+    # as the current implementation of pointwise does only support CUDA device
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
```

Comment:
We might just remove the parametrization of combine_mode

Reply:
Done